x-pack/legacy/plugins/ml/public/overview/components/anomaly_detection_panel/utils.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { JOB_STATE, DATAFEED_STATE } from '../../../../common/constants/states';
+import { Group, Groups, Jobs, Job } from './anomaly_detection_panel';
+
+export function getGroupsFromJobs(jobs: Jobs): Groups {
+  const groups: any = {
+    ungrouped: {
+      id: 'ungrouped',
+      jobIds: [],
+      docs_processed: 0,
+      latest_timestamp: 0,
+      max_anomaly_score: null,
+    },
+  };
+
+  jobs.forEach((job: any) => {
```

Comment:
there is a Job interface at `x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_creator/configs/job.ts`